### PR TITLE
Refactor CollectionViewSources

### PR DESCRIPTION
**Pull request by willsb**
Needed for a future PR

Refactored a lot of copy pasta into a single class. Much better now.

Squash ASAP

### CODE REVIEW

File: Toggl.Daneel/ViewSources/SelectProjectTableViewSource.cs [hunk 1]
```diff
@@ -11,60 +11,30 @@
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/SelectProjectTableViewSource.cs]: `using`s on lines 5 and 5 have become unnecessary.

File: Toggl.Daneel/ViewSources/SelectProjectTableViewSource.cs [hunk 2]
```diff
@@ -11,60 +11,30 @@
...
-            => getGroupAt(section).Count();
-
-        private WorkspaceGroupedCollection<ProjectSuggestion> getGroupAt(nint section)
-            => groupedItems.ElementAt((int)section);
-
-        protected override object GetItemAt(NSIndexPath indexPath)
-            => groupedItems.ElementAt(indexPath.Section).ElementAt((int)indexPath.Item);
-
-        private UITableViewHeaderFooterView getOrCreateHeaderViewFor(UITableView tableView)
+        protected override UITableViewHeaderFooterView GetOrCreateHeaderViewFor(UITableView tableView)
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/SelectProjectTableViewSource.cs]: Could the `GetOrCreateCellFor` and `GetOrCreateHeaderViewFor` methods be moved to the base class if `headerCellIdentifier` and `cellidentifier` are moved to the base class and made abstract?

File: Toggl.Daneel/ViewSources/SelectTagsTableViewSource.cs [hunk 3]
```diff
@@ -11,60 +11,28 @@
```
Comment (arturs105) [inline Toggl.Daneel/ViewSources/SelectTagsTableViewSource.cs]: Some `using`s are also unnecessary.

File: Toggl.Daneel/ViewSources/SelectProjectTableViewSource.cs (hunk 2 above)
Comment (willsb) [reply Toggl.Daneel/ViewSources/SelectProjectTableViewSource.cs]: Good point. Some might need overriding, but most won't. WIll do

**Pull request merged**